tests/saw/bike/spec/abstract_spec/algorithm3.cry

```diff
@@ -0,0 +1,169 @@
+// algorithm3.cry Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+///////////////////////////////////////////////////////////////////////////////
+
+module abstract_spec::algorithm3 where
+
+import abstract_spec::vectors
+import abstract_spec::vector_sets
+import abstract_spec::ring
+import abstract_spec::types
+import abstract_spec::parameters as P
+import abstract_spec::random_choice
+import lib::Maybe
+
+///////////////////////////////////////////////////////////////////////////////
+// BIKE Algorithm 3 "One-Round Bit Flipping Algorithm", from Section 2.5.1
+
+// NOTE: The algorithm description there has several loops and a few
+//       subfunctions.  We translate the while loops into recursive
+//       functions. For loops can be translated into Cryptol comprehensions.
+
+threshold_fn _ = 2
+
+type syndrome = R_vector P::r
+type J_set = vector P::n // vector-as-set
+type pk_matrix = matrix P::r P::n
+
+// NOTE: The spec uses `d/2`, but d can be odd.  Absent a module for rationals
+//       in Crytpol, which seems like overkill, not that the only context in
+//       which d/2 is used is in expressions like "e >= d/2", where e is an
+//       integer. So if we use division with rounding up, we will get the
+//       correct result.
+half_rounded_up: Integer -> Integer
+half_rounded_up x = (1+x)/2
+
+// TODO: How many Js are there?  The subscript is never larger than the
```

Comment:
Going through all the `TODO` comments in this PR, does this question still need to be answered?